src/dune_rules/dir_contents.ml

```diff
@@ -180,17 +180,28 @@ end = struct
           | Generate_sites_module def ->
             let+ res = Generate_sites_module_rules.setup_rules sctx ~dir def in
             [ res ]
-          | Library { buildable; melange_runtime_deps; _ } ->
+          | Library { buildable; melange_runtime_deps; modes; name; _ } ->
             let files = buildable_files buildable in
             let* melange_files =
               match melange_runtime_deps with
               | None -> Memo.return []
-              | Some files ->
-                let+ ps =
-                  Simple_rules.extra_files_melange sctx files ~src_dir ~dir
-                    ~expander
-                in
-                Path.Set.to_list_map ps ~f:Path.basename
+              | Some (loc, files) -> (
+                match modes.melange with
+                | None ->
+                  User_error.raise ~loc
+                    [ Pp.textf
+                        "Library %S is using \"melange.runtime_deps\", but it \
+                         is not a Melange library. To fix this error, you must \
+                         add \"melange\" to \"modes\", or remove \
+                         \"melange.runtime_deps\"."
```

Comment:
I just checked the other errors in our code, and we don't quote field names. I'd drop the quotes here for the same reason. By the way, can this validation be moved to Dune_file? We should do it as early as possible.

Reply:
I don't think there is a lot of consistency, there are cases [without quotes](https://github.com/ocaml/dune/blob/c016117ee26ca22c04123e87aa4394b554e22e95/src/dune_rules/melange_stanzas.ml#L67), with [single quotes](https://github.com/ocaml/dune/blob/c016117ee26ca22c04123e87aa4394b554e22e95/src/dune_rules/dune_file.ml#L1819) and with [double quotes](https://github.com/ocaml/dune/blob/c016117ee26ca22c04123e87aa4394b554e22e95/src/dune_rules/foreign.ml#L187).

Happy to unify them in another PR.

For now, removed the quotes in this case and moved to `dune_file` in https://github.com/ocaml/dune/pull/6643/commits/4a7e8b5a21c19dfc880df13acab59da5b61bdaa4.